src/webots/control/WbController.cpp

```diff
@@ -887,6 +883,25 @@ const QString &WbController::args() const {
   return mRobot->controllerArgs();
 }
 
+QStringList WbController::argsList() const {
+  QStringList list;
+  const QString args = mRobot->controllerArgs().trimmed();
+  if (args.length() == 0)
+    return list;
+  bool quote = false;
+  int previous = 0;
+  for (int i = 0; i < args.length(); i++) {
+    if (args[i] == '"')
+      quote = !quote;
+    if (args[i] == ' ' && !quote) {
+      list << args.mid(previous, i - previous - 1).replace("\"", "");
```

Comment:
```suggestion
      const QString argument = args.mid(previous, i - previous).replace("\"", "").trimmed();
      if (!argument.isEmpty())
        list << args.mid(previous, i - previous).replace("\"", "");
```

The problem was indeed not the removed quotes, but there were actually to issues on this line:
  1. if more than one space was used between arguments, then some empty arguments were pushed in the list.
  2. the cutting of the arguments was wrong. This was actually visible in the starting sequence I send before (but I just saw it know): `... --initial-linear-velocity= --initial-angular-velocity=0. --are...` (last character of each argument is missing).